### PR TITLE
WIP: file collection after deprecation of 'py' in pytest

### PR DESCRIPTION
**Pull request by SirUli**
pytest-molecule suffers from one of the [deprecations](https://docs.pytest.org/en/7.0.x/changelog.html#id20):

- https://github.com/pytest-dev/pytest/issues/7259
- https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
- https://docs.pytest.org/en/7.0.x/deprecations.html#legacy-path-hooks-deprecated

### CODE REVIEW

**Review by ssbarnea — CHANGES_REQUESTED** [review]
It seems incomplete. Can you also add type information while you modify it?

**Comment by SirUli** [comment]
@ssbarnea I hope this fits your expectation

**Comment by ssbarnea** [comment]
> @ssbarnea I hope this fits your expectation

It is getting close! mypy still fails. Once ready I will merge it and make a release. I don't want to prevent others from using pytest 7.x

File: src/pytest_molecule/__init__.py
Comment (SirUli) [inline src/pytest_molecule/__init__.py]: @ssbarnea i have two possibilities to fix that mypy finding as per my understanding:

```
src/pytest_molecule/__init__.py:138: error: Argument 1 to "MoleculeFile" has incompatible type "Path"; expected "Optional[local]"
```

1) Explicitly only allow pytest greater 5.4 to have "from_parent" available which allows us to simplify these three lines 136-138 to:

```
return MoleculeFile.from_parent(path=file_path, parent=parent)
```

5.4 was released March 2020.

2) We retain compatibility with pytest < 5.4 through an conditional import of py.path.local to convert the file_path from Pathlib.Path to py.path.local.

Which route do you want to go?

Thanks in advance!

Best Regards, Uli

Comment (ssbarnea) [reply src/pytest_molecule/__init__.py]: Bump minimum pytest version requires to 5.4. That release is more than two years old. If someone wants to use older ones they should fork it ;)

Comment (SirUli) [reply src/pytest_molecule/__init__.py]: Actually it gets a little worse. Due to the changed method definition (especially the typing) of pytest_collect_file pytest<7 doesn't like it anymore (I only tested against pytest 7 so far so didn't come across that). I'll check tomorrow how the compatibility with pytest 6 could be retained - I have an idea.

File: src/pytest_molecule/__init__.py
Comment (SirUli) [inline src/pytest_molecule/__init__.py]: mypy complains here with:

```
src/pytest_molecule/__init__.py:142: error: All conditional function variants must have identical signatures
```

@ssbarnea this is intentional since they actually are different in the pytest versions. Ideas? I tend to ignore the error but maybe you have a better idea for the implementation?

Thanks in advance!

**Comment by ssbarnea** [comment]
I would just drop support for older pytest versions and require 7+. I personally do not have the bandwidth to keep compatibility with older versions.

**Comment by SirUli** [comment]
@ssbarnea i dropped the support for pytest < 7. Sorry for the delay - went on parental leave and forgot about that being open ;)

**Comment by ssbarnea** [comment]
Closing as #121 is implementing it. I tried to update your PR but it was created without permissions for maintainers to update it, so I was forced to create a new one.

Thanks for taking time to do it!

**Comment by SirUli** [comment]
I actually didn't intentionally choose to do so. Perfect - thanks for your help.

Sorry for the lengthy process - i learned quite a bit during it and hope to better contribute to upcoming changes ;)

**Comment by ssbarnea** [comment]
@SirUli I hope so. With community projects we never know how long it would take to ship a change. I will be happy to see more PRs from you.

**Pull request closed without merging**
